src/agent/protocols/hack/update-generated-proto.sh

```diff
@@ -1,68 +1,133 @@
+        sed 's/fn into_any(self: Box<Self>) -> ::std::boxed::Box<::std::any::Any> {/fn into_any(self: ::std::boxed::Box<Self>) -> ::std::boxed::Box<::std::any::Any> {/g' ./protocols/src/oci.rs > ./protocols/src/new_oci.rs
+        sed 's/fn into_any(self: Box<Self>) -> ::std::boxed::Box<dyn (::std::any::Any)> {/fn into_any(self: ::std::boxed::Box<Self>) -> ::std::boxed::Box<dyn (::std::any::Any)> {/g' ./protocols/src/oci.rs > ./protocols/src/new_oci.rs
+        mv ./protocols/src/new_oci.rs ./protocols/src/oci.rs
+    fi;
+}
+
+if [ "$(basename $(pwd))" != "agent" ]; then
 	die "Please go to directory of protocols before execute this shell"
 fi
+
+if [ "X$1" = "X" ]; then
+    show_usage
+    exit 1
+fi;
+
+# pre-requirement check
 which protoc
 [ $? -eq 0 ] || die "Please install protoc from github.com/protocolbuffers/protobuf"
+
 which protoc-gen-rust
 [ $? -eq 0 ] || die "Please install protobuf-codegen from github.com/pingcap/grpc-rs"
-which grpc_rust_plugin
-[ $? -eq 0 ] || die "Please install grpc_rust_plugin from github.com/pingcap/grpc-rs"
-
-if [ $UPDATE_PROTOS ]; then
-    if [ ! $GOPATH ]; then
-        die 'Need $GOPATH to get the proto files'
-    fi
-
-    get_source_version "github.com/kata-containers/agent" ""
-    cp $GOPATH/src/github.com/kata-containers/agent/protocols/grpc/agent.proto ./protos/
-    cp $GOPATH/src/github.com/kata-containers/agent/protocols/grpc/oci.proto ./protos/
-    cp $GOPATH/src/github.com/kata-containers/agent/protocols/grpc/health.proto ./protos/
-    mkdir -p ./protos/github.com/kata-containers/agent/pkg/types/
-    cp $GOPATH/src/github.com/kata-containers/agent/pkg/types/types.proto ./protos/github.com/kata-containers/agent/pkg/types/
-
-    # The version is get from https://github.com/kata-containers/agent/blob/master/Gopkg.toml
-    get_source_version "github.com/gogo/protobuf" "4cbf7e384e768b4e01799441fdf2a706a5635ae7"
-    mkdir -p ./protos/github.com/gogo/protobuf/gogoproto/
-    cp $GOPATH/src/github.com/gogo/protobuf/gogoproto/gogo.proto ./protos/github.com/gogo/protobuf/gogoproto/
-    mkdir -p ./protos/google/protobuf/
-    cp $GOPATH/src/github.com/gogo/protobuf/protobuf/google/protobuf/empty.proto ./protos/google/protobuf/
-fi
 
-get_rs agent.proto
-get_rs health.proto
-get_rs github.com/kata-containers/agent/pkg/types/types.proto
-get_rs google/protobuf/empty.proto
+which ttrpc_rust_plugin
+[ $? -eq 0 ] || die "Please install ttrpc_rust_plugin from https://github.com/containerd/ttrpc-rust"
+
+which protoc-gen-gogottrpc
+[ $? -eq 0 ] || die "Please install protoc-gen-gogottrpc from https://github.com/containerd/ttrpc"
+
+golang_files=(agent.proto health.proto oci.proto github.com/kata-containers/agent/pkg/types/types.proto)
+rust_files=(agent.proto health.proto oci.proto github.com/kata-containers/agent/pkg/types/types.proto google/protobuf/empty.proto)
+
+# do generate work
+target=$1
+
+# compile all proto files
+if [ "X$target" = "Xall" ]; then
```

Comment:
I think these tests can be simplified to avoid the `X` prefix - wasn't that used a long time ago to handle different shell semantics? Either way, I don't think it's required these days as we use `bash` everywhere :smile:

Reply:
Yes, you are right